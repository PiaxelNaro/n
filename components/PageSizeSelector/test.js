import React from 'react';
import { shallow, mount } from 'enzyme';
import { _PageSizeSelector as PageSizeSelector } from '.';
import mockI18n from '../../modules/i18n/mockI18n';

describe('<PageSizeSelector>', () => {
    it('renders PageSizeSelector with default options', () => {
        expect(
            shallow(<PageSizeSelector current={100} i18n={mockI18n} />)
        ).toMatchSnapshot();
    });

    it('renders PageSizeSelector with customized options', () => {
        const spy = jest.fn();
        const select = mount(
            <PageSizeSelector
                options={[10, 50, 100]}
                current={10}
                onSelectionChanged={spy}
                i18n={mockI18n}
            />
        );
        expect(select).toMatchSnapshot();

        const value = 50;
        select.find('select').simulate('change', {
            target: {
                value,
            },
        });
        expect(spy).toHaveBeenCalledWith(value);
    });
});
