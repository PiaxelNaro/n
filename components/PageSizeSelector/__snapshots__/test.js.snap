// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageSizeSelector> renders PageSizeSelector with customized options 1`] = `
<PageSizeSelector
  current={10}
  i18n={
    Object {
      "gettext": [Function],
      "lang": "en-US",
      "ngettext": [Function],
      "npgettext": [Function],
      "pgettext": [Function],
    }
  }
  onSelectionChanged={[Function]}
  options={
    Array [
      10,
      50,
      100,
    ]
  }
>
  <PageSizeSelector__Wrapper>
    <div
      className="PageSizeSelector__Wrapper-s8km4x9-0 jtXQfE"
    >
      Show rows:
      <PageSizeSelector__RawSelect
        onChange={[Function]}
        value={10}
      >
        <select
          className="PageSizeSelector__RawSelect-s8km4x9-1 hpABCI"
          onChange={[Function]}
          value={10}
        >
          <option
            value={10}
          >
            10
          </option>
          <option
            value={50}
          >
            50
          </option>
          <option
            value={100}
          >
            100
          </option>
        </select>
      </PageSizeSelector__RawSelect>
    </div>
  </PageSizeSelector__Wrapper>
</PageSizeSelector>
`;

exports[`<PageSizeSelector> renders PageSizeSelector with default options 1`] = `
<PageSizeSelector__Wrapper>
  Show rows:
  <PageSizeSelector__RawSelect
    onChange={[Function]}
    value={100}
  >
    <option
      value={10}
    >
      10
    </option>
    <option
      value={50}
    >
      50
    </option>
    <option
      value={100}
    >
      100
    </option>
    <option
      value={500}
    >
      500
    </option>
    <option
      value={1000}
    >
      1000
    </option>
  </PageSizeSelector__RawSelect>
</PageSizeSelector__Wrapper>
`;
